### PR TITLE
189/Simplify filenamefactory

### PR DESCRIPTION
**Pull request by friedger**
Depends on PR #205 

Just remove unused methods so that the purpose of the factory is clear

### CODE REVIEW

File: core/src/main/java/com/novoda/pxfetcher/HttpResourceManager.java [hunk 1]
```diff
@@ -15,7 +12,9 @@
 import java.net.HttpURLConnection;
 import java.net.URL;
 
-public class UrlNetworkManager implements NetworkManager {
+public class HttpResourceManager implements ResourceManager {
+
+    private static final String TAG = "UrlNetworkManager";
```
Comment (ataulm) [inline core/src/main/java/com/novoda/pxfetcher/HttpResourceManager.java]: Update tag? `HttpResourceManager.class.getSimpleName()` to protect against refactorings?

Comment (xrigau) [inline core/src/main/java/com/novoda/pxfetcher/HttpResourceManager.java]: :+1: @ataulm

File: core/src/main/java/com/novoda/pxfetcher/ResourceManager.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2012 Novoda Ltd
```
Comment (ataulm) [inline core/src/main/java/com/novoda/pxfetcher/ResourceManager.java]: Can we kill these? Or are they a necessity? The license is on the README already.

File: core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java [hunk 3]
```diff
@@ -4,10 +4,4 @@
 
     String getFileName(String sourceUrl, T metadata);
 
-    T extractMetadata(String fileName);
-
-    boolean hasMetadata(String file);
-
-    String getCacheDirectory();
```
Comment (ataulm) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: :x: this was added out of needed in 2 projects - @Dorvaryn and @mr-archano can comment more (more Ben as he added it)

File: core/build.gradle [hunk 4]
```diff
@@ -26,7 +18,7 @@ android {
         minSdkVersion androidMinSdkVersionProp
         targetSdkVersion androidTargetSdkVersionProp
         versionCode Integer.valueOf(androidVersionCodeProp)
-        versionName androidVersionNameProp
+        versionName version
```
Comment (ataulm) [inline core/build.gradle]: where does `version` come from?

Comment (ouchadam) [inline core/build.gradle]: I don't think `versionName` or `versionCode` are needed for a android library

File: core/src/main/java/com/novoda/pxfetcher/ResourceManager.java (hunk 2 above)
Comment (ouchadam) [inline core/src/main/java/com/novoda/pxfetcher/ResourceManager.java]: :100:

File: core/build.gradle (hunk 4 above)
Comment (friedger) [inline core/build.gradle]: version comes from build.gradle. There used to be androidVersionNameProp but that was not maintained when upping to 2.0-BETA

File: core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java (hunk 3 above)
Comment (friedger) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: The names of the removed methods suggest that these methods are not related to the purpose of the file name factory

Comment (friedger) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: @ataulm Let's move the extension of the FileNameFactory API to a new PR.

Comment (friedger) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: @ataulm Save in https://github.com/novoda/image-loader/issues/207

Comment (ataulm) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: Just pinged @Dorvaryn to see this, but :shipit: - I'm happy to have this as a separate PR/issue.

File: core/build.gradle (hunk 4 above)
Comment (ataulm) [inline core/build.gradle]: :+1: remove if not needed?

File: core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java (hunk 3 above)
Comment (Dorvaryn) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: Why are we removing this to add it again in #207 then ?

The idea of this is to be able to index metadata as part of the filename to query localy cached images depending on those metadata.

This allow for usage accross app/widget/daydream with pre caching of the images relevant to daydream and/or of a certain type/category.

Comment (ataulm) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: Probably it just needed better names. FileNameFactory allowed the user to
encode metadata into a filename, and decode from the filename, where the
metadata is of type T.

On 28 January 2015 at 15:28, Benjamin AUGUSTIN <notifications@github.com>
wrote:

> In core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java
> <https://github.com/novoda/image-loader/pull/206#discussion_r23694280>:
>
> > @@ -4,10 +4,4 @@
> >
> >      String getFileName(String sourceUrl, T metadata);
> >
> > -    T extractMetadata(String fileName);
> > -
> > -    boolean hasMetadata(String file);
> > -
> > -    String getCacheDirectory();
>
> Why are we removing this to add it again in #207
> <https://github.com/novoda/image-loader/issues/207> then ?
>
> The idea of this is to be able to index metadata as part of the filename
> to query localy cached images depending on those metadata.
>
> This allow for usage accross app/widget/daydream with pre caching of the
> images relevant to daydream and/or of a certain type/category.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/novoda/image-loader/pull/206/files#r23694280>.
>

-- 
Disclaimer: The information in this e-mail and any attachments is the 
property of Novoda Ltd and is confidential and may be legally privileged. 
This e-mail is intended solely for the person or organisation to which it 
is addressed. Any disclosure, copying or other use of the information by 
any person or organisation who is not the intended recipient is strictly 
prohibited and may be unlawful. If you have received this e-mail in error, 
please inform the sender immediately and delete/destroy this e-mail and any 
copies of it. Novoda Ltd has taken reasonable precautions to minimise the 
risk of any software viruses which may damage your systems, but we advise 
that you take the necessary steps to ensure that no virus contamination is 
suffered. Novoda Ltd does not accept any liability for any loss or damage 
caused by the transmission of any virus.

Novoda Ltd, Company No: 347444, Registered in Scotland Registered 
Office: C/O Alexander Sloan, 38 Cadogan Street, Glasgow, G2 7HF, Scotland.  VAT 
Registration Number  GB 984 2525 93

File: core/build.gradle (hunk 4 above)
Comment (blundell) [inline core/build.gradle]: yeah they're not needed :+1: removeable

File: core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java (hunk 3 above)
Comment (blundell) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: @friedger if it's a public interface, how could you determine they're unused? Guess now Ben/Ataul are saying it was used!

Comment (friedger) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: @blundell I didn't say where they were not used. Ignoring your comment ;-)

If I understand the purpose of the methods correctly I would move the API to the CacheManager or the specific Retriever.

Comment (blundell) [inline core/src/main/java/com/novoda/pxfetcher/FileNameFactory.java]: This is probably a good place to organise a quick hangout between the people that know (not me!) @ataulm @Dorvaryn @friedger ?
